feat: adding the OOP platformer project to new JS curriculum (#50385)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64ab178206f3237eafcc0ef4.md

---
id: 64ab178206f3237eafcc0ef4
title: Step 37
challengeType: 0
dashedName: step-37
---

# --description--

The <dfn>requestAnimationFrame()</dfn> web API, takes in a callback and is used to update the animation on the screen. The `animate` function will be responsible for updating the player's position and continually drawing it on the canvas. 

> Inside the `animate` function, call the `requestAnimationFrame()` API and pass in `animate` for the argument.

Reviewer: ```suggestion
Inside the `animate` function, call the `requestAnimationFrame()` API and pass `animate` as the argument.
```
